chainindex/indexer.go

```diff
@@ -62,6 +68,9 @@ type SqliteIndexer struct {
 
 	closeLk sync.RWMutex
 	closed  bool
+
+	// ensures writes are serialized so backfilling does not race with index updates
+	writerLk sync.Mutex
```

Comment:
@rvagg This is the big change here. This prevents a race between `Apply/Revert` and backfilling. `Apply/Reverts` are serialised by the `Observer` but we could end up over-writing a `Revert` with a backfill.